adapters/repos/db/migrator.go

```diff
@@ -334,8 +334,32 @@ func (m *Migrator) UpdateTenants(ctx context.Context, class *models.Class, updat
 	ec := &errorcompounder.ErrorCompounder{}
 
 	for _, name := range updatesHot {
-		_, err := idx.getOrInitLocalShard(ctx, name)
+		shard, err := idx.getOrInitLocalShard(ctx, name)
 		ec.Add(err)
+		if err != nil {
+			continue
+		}
+
+		// if the shard is a lazy load shard, we need to force its activation now
+		asLL, ok := shard.(*LazyLoadShard)
+		if !ok {
+			continue
+		}
+
+		go func(name string) {
```

Comment:
**suggestion (non-blocking):** shall we use the internal wrapper, wdyt ?

Reply:
That is a great point, let me push this change.